content/articles/2018-06-conflicted.Rmd

```diff
@@ -0,0 +1,86 @@
+---
+title: "conflicted: a new approach to resolving ambiguity"
+author: Hadley Wickham
+date: '2018-06-02'
+slug: conflicted
+categories:
+  - package
+photo:
+  url: https://unsplash.com/photos/M03OCZvYSxY
+  author: Vincent van Zalinge
+tags: []
+---
+
+```{r}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+
+We're stoked to announce the initial release of the [conflicted](https://github.com/r-lib/conflicted#conflicted) package. The goal of conflicted is to provide alternative way of resolving conflicts caused by ambiguous function names. R handles ambiguity by reporting conflicts when you load a package, then having the most recently loaded package win. This can make it hard to detect conflicts, because it's easy to miss the messages since you often load packages at the top of the script, and you don't see a problem until much later. conflicted takes a different approach to resolving ambiguity, instead making every conflict an error and forcing you to always choose which function to use.
+
+Installed conflicted by running:
+
+```{r, eval = FALSE}
+install.packages("conflicted")
+```
+
+## How do I use it?
+
+conflicted does not export any functions. To use it, you just need to load it:
+
+```{r, error = TRUE}
+library(conflicted)
+library(dplyr)
+
+filter(mtcars, am & cyl == 8)
+```
+
+To resolve conflicts, remove the ambiguity by using `::` (i.e. `dplyr::filter()` or `base::filter()`). To resolve conflicts for your entire session, use `<-`:
+
+```{r}
+filter <- dplyr::filter
+filter(mtcars, am & cyl == 8)
+```
+
+If you want to make this behaviour the default, you can load conflicted in your `~/.Rprofile` (the easiest way to find and edit this file is with `usethis::edit_r_profile()`):
+
+```{r}
+if (interactive()) {
+  require(conflicted)
+}
+```
+
+## How does it work?
+
+Loading conflicted creates a new "conflicted" environment that is attached just after the global environment. This environment contains an active bindings for any object that is exported by multiple packages; the active binding will throw an error message describing how to disambiguate the name.
+
+The conflicted environment also contains bindings for `library()` and `require()`, these  suppress the default conflict reporting (i.e. `warn.conflicts = FALSE`) and ensure that the conflicted environment is updated with any new conflicts. Because they occur in the search path before the base package, they will be called instead of `base::library()` and `base::require()`.   
```

Comment:
Either change these to which (these suppress the default...), or use a semicolon or start a new sentence at "these suppress"